### DISCUSSION TITLE
Filter based on resolved field on a `prismaNode`

### DISCUSSION THREAD

**Discussion by FarazPatankar** [hayes/pothos#1156] [Q&A]
Hey, possibly a tricky one but curious if this is possible.
```ts
// Template node
builder.prismaNode("Template", {
  id: { field: "id" },
  fields: t => ({
    ...BASIC_MODEL_FIELDS
    health: t.float({ resolve: () => { DYNAMIC_VALUE } })
  })
})

builder.queryFields(t => ({
    templates: t.prismaConnection({
      ...BASIC_FIELDS
      resolve: () => {
        // Fetch data from DB and return
      }
    })
  })
)
```

Above is a basic snippet of what I have + concerned fields.
1. The `health` field is calculated on the fly. 
2. The `templates` query resolver fetches a bunch of templates from the DB and returns them.
3. AFAIK, I cannot access and filter on the value of the `health` field inside the `resolve` fn for the query.

Is there a way to tell Pothos not to return a particular template in the response based on the value of the `health` field? I could do a simple filter on the client but preference would be to filter on the server and never return certain objects.

**Comment by hayes** [comment]
No, that's not possible.  There is no way to resolve a child field before a parent is fully resolved.  This just doesn't make sense, you have a health field for each template, and your resolver is passed the template as the parent object even if you don't use it.   You need to fetch your list of templates before fetching your templates.  If you don't need the template to determine health, you can extract the logic and re-use it in the query resolver instead

**Reply by FarazPatankar** [reply]
> If you don't need the template to determine health

Sadly we do. Wonder if this warrants storing the health in the DB once we've calculated it as that'll make things much more straightforward.

Thanks for the response. 😄
